docs/spec/lightclient/detection/req-ibc-detection.md

```diff
@@ -0,0 +1,296 @@
+validators of some smaller height.
+
+In principle everyone can detect a fork
+
+- ./detection talks about the Tendermint light client with a focus on 
+  light nodes.
+  
+- a relayer runs such light clients and may detect forks in this way
+
+- in IBC, a relayer can see that a handler is on a conflicting branch
+    - the relayer should feed the handler the necessary information so
+      that it can halt
+	- the relayer should report the fork to a full node
+	
+### Evidence creation and submission
+
+- the information sent from the relayer to the handler could be called
+  evidence, but this is perhaps a bad idea because the information sent to a
+  full node can also be called evidence. But this evidence might still
+  not be enough as the full node might need to run the "fork
+  accountability" protocol to generate evidence in the form of
+  consensus messages. So perhaps we should
+  introduce different terms for:
+  
+  - proof of fork for the handler (basically consisting of lightblocks)
+  - proof of fork for a full node (basically consisting of (fewer) lightblocks)
+  - proof of misbehavior (consensus messages)
+  
+
+
+### Isolating misbehaving nodes
+
+- this is the job of a full node. 
+
+- might be subjective in the future: the protocol depends on what the
+  full node believes is the "correct" chain. Right now we postulate
+  that every full node is on the correct chain, that is, there is no
+  fork on the chain.
+  
+- The full node figures out which nodes are
+    - lunatic
+	- double signing
+	- amnesic; **using the challenge response protocol**
+
+- We do not punish "phantom" validators
+    - currently we understand a phantom validator as a node that
+	    - signs a block for a height in which it is not in the
+          validator set
+        - the node is not part of the +1/3 of previous validators that
+          are used to support the header. Whether we call a validator
+          phantom might be subjective and depend on the header we
+          check against. Their formalization actually seems not so
+          clear.
+	- they can only do something if there are +1/3 faulty validators
+      that are either lunatic, double signing, or amnesic.
+    - abci requires that we only report bonded validators. So if a
+      node is a "phantom", we would need the check whether the node is
+      bonded, which currently is expensive, as it requires checking
+      blocks from the last three weeks.
+	- in the future, with state sync, a correct node might be
+      convinced by faulty nodes that it is in the validator set. Then
+      it might appear to be "phantom" although it behaves correctly
+
+
+## Next steps
+
+> The following points are subject to my limited knowledge of the
+> state of the work on IBC. Some/most of it might already exist and we
+> will just need to bring everything together.
+
+- "proof of fork for a full node" defines a clean interface between
```

Comment:
In my view these are the components for which we need precise specs: 1) detecting forks on IBC module, 2) proof of fork submission and verification (most probably separate protocols for full node and IBC module), 3) creating evidence(s) from a valid proof of fork (executed by a full node). We also need to specify 4) evidence gossip and on chain commitment and 2) on chain evidence handling with amnesia handling requiring more complex challenge-response + fork accountability protocol execution. In terms of priority, 1), 2) and 3) have higher priority, as 4) and 5) are in some fork already specified before and it shouldn't be affected by 1)-3).